csrc/kernel_cache.cpp

```diff
@@ -755,6 +754,44 @@ std::unordered_map<Val*, const ArgAbstract*> FusionKernelRuntime::
       " inputs but expected ",
       segmented_fusion_->inputs().size());
 
+  // map val to segment_id where arg is lastly used
```

Comment:
Please move this part out of this function

Reply:
I moved to a new class `ArgumentManager`